fms/models/__init__.py

```diff
@@ -81,30 +81,54 @@ def __maybe_infer_model_variant(
     if architecture in ("hf_pretrained", "hf_configured"):
         from fms.models.hf.utils import _infer_model_configuration  # type: ignore
 
-        logger.info(f"inferring model configuration from {variant}")
         is_hf_pretrained = architecture == "hf_pretrained"
+        is_hf_configured = architecture == "hf_configured"
+
+        if is_hf_pretrained:
+            if variant is None:
+                model_path_or_variant = model_path  # type: ignore[assignment]
+            else:
+                model_path_or_variant = variant
+        elif is_hf_configured:
+            model_path_or_variant = variant
+
         if is_hf_pretrained:
-            if model_path is not None or source is not None:
+            if (variant is None == model_path is None) or source is not None:
                 raise ValueError(
-                    """architecture="hf_pretrained" implies model weights will be downloaded and extracted from hf cache and loaded into the model, therefore model_path and source should not be set"""
+                    """
+                    architecture="hf_pretrained" implies one of two things: 
+                    1. if variant is defined, model config and weights will be downloaded and extracted from hf cache and loaded into the model, therefore model_path should not be set.
```

Comment:
maybe add here that downloading the weights only occurs if not already in the hf cache